File: app/services/test_data.py
"""Test data constants for football API fixtures."""

MOCK_CLUBS = [
    {"id": 1, "name": "Real Madrid", "league_id": 1, "league_name": "La Liga"},
    {"id": 2, "name": "Barcelona", "league_id": 1, "league_name": "La Liga"},
    {"id": 3, "name": "Atletico Madrid", "league_id": 1, "league_name": "La Liga"},
    {"id": 4, "name": "Sevilla", "league_id": 1, "league_name": "La Liga"},
    {"id": 5, "name": "Valencia", "league_id": 1, "league_name": "La Liga"},
    {
        "id": 6,
        "name": "Manchester United",
        "league_id": 2,
        "league_name": "Premier League",
    },
    {"id": 7, "name": "Liverpool", "league_id": 2, "league_name": "Premier League"},
    {
        "id": 8,
        "name": "Manchester City",
        "league_id": 2,
        "league_name": "Premier League",
    },
    {"id": 9, "name": "Chelsea", "league_id": 2, "league_name": "Premier League"},
    {"id": 10, "name": "Arsenal", "league_id": 2, "league_name": "Premier League"},
    {"id": 11, "name": "Bayern Munich", "league_id": 3, "league_name": "Bundesliga"},
    {
        "id": 12,
        "name": "Borussia Dortmund",
        "league_id": 3,
        "league_name": "Bundesliga",
    },
    {"id": 13, "name": "RB Leipzig", "league_id": 3, "league_name": "Bundesliga"},
    {"id": 14, "name": "Bayer Leverkusen", "league_id": 3, "league_name": "Bundesliga"},
    {
        "id": 15,
        "name": "Eintracht Frankfurt",
        "league_id": 3,
        "league_name": "Bundesliga",
    },
    {"id": 16, "name": "Juventus", "league_id": 4, "league_name": "Serie A"},
    {"id": 17, "name": "AC Milan", "league_id": 4, "league_name": "Serie A"},
    {"id": 18, "name": "Inter Milan", "league_id": 4, "league_name": "Serie A"},
    {"id": 19, "name": "Napoli", "league_id": 4, "league_name": "Serie A"},
    {"id": 20, "name": "Roma", "league_id": 4, "league_name": "Serie A"},
    {"id": 21, "name": "Inter Miami", "league_id": 5, "league_name": "MLS"},
    {"id": 22, "name": "Al-Nassr", "league_id": 6, "league_name": "Saudi Pro League"},
]

MOCK_NATIONS = [
    {"id": 1, "name": "Spain", "country_code": "ESP"},
    {"id": 2, "name": "Brazil", "country_code": "BRA"},
    {"id": 3, "name": "Argentina", "country_code": "ARG"},
    {"id": 4, "name": "France", "country_code": "FRA"},
    {"id": 5, "name": "Germany", "country_code": "DEU"},
    {"id": 6, "name": "Italy", "country_code": "ITA"},
    {"id": 7, "name": "England", "country_code": "ENG"},
    {"id": 8, "name": "Portugal", "country_code": "POR"},
    {"id": 9, "name": "Netherlands", "country_code": "NED"},
    {"id": 10, "name": "Uruguay", "country_code": "URU"},
]

MOCK_LEAGUES = [
    {"id": 1, "name": "La Liga"},
    {"id": 2, "name": "Premier League"},
    {"id": 3, "name": "Bundesliga"},
    {"id": 4, "name": "Serie A"},
    {"id": 5, "name": "MLS"},
    {"id": 6, "name": "Saudi Pro League"},
]

MOCK_POSITIONS = [
    {"code": "GK", "name": "Goalkeeper"},
    {"code": "DEF", "name": "Defender"},
    {"code": "MID", "name": "Midfielder"},
    {"code": "FWD", "name": "Forward"},
]

File: football-cards-ui/tests/e2e/base/fixtures/test-data.ts
/**
 * Test data fixtures for E2E tests
 */

// Sample player data
export const SAMPLE_PLAYERS = {
  CRISTIANO_RONALDO: {
    name: 'Cristiano Ronaldo',
    club: 'Real Madrid',
    nationality: 'Portugal',
    league: 'La Liga',
    position: 'Forward',
    stats: { defence: 45, control: 88, attack: 95 },
  },
  LIONEL_MESSI: {
    name: 'Lionel Messi',
    club: 'Barcelona',
    nationality: 'Argentina',
    league: 'La Liga',
    position: 'Forward',
    stats: { defence: 35, control: 92, attack: 94 },
  },
  TEST_PLAYER: {
    name: 'Test Player',
    club: 'Arsenal',
    nationality: 'England',
    league: 'Premier League',
    position: 'Midfielder',
    stats: { defence: 75, control: 80, attack: 70 },
  },
};

// Sample clubs data
export const SAMPLE_CLUBS = [
  'Real Madrid',
  'Barcelona',
  'Atletico Madrid',
  'Sevilla',
  'Valencia',
  'Manchester United',
  'Liverpool',
  'Manchester City',
  'Chelsea',
  'Arsenal',
  'Bayern Munich',
  'Borussia Dortmund',
  'RB Leipzig',
  'Bayer Leverkusen',
  'Eintracht Frankfurt',
  'Juventus',
  'AC Milan',
  'Inter Milan',
  'Napoli',
  'Roma',
];

// Sample nationalities
export const SAMPLE_NATIONALITIES = [
  'Spain',
  'Brazil',
  'Argentina',
  'France',
  'Germany',
  'Italy',
  'England',
  'Portugal',
  'Netherlands',
  'Uruguay',
];

// Sample leagues
export const SAMPLE_LEAGUES = [
  'La Liga',
  'Premier League',
  'Bundesliga',
  'Serie A',
];

// Sample positions
export const SAMPLE_POSITIONS = [
  'Goalkeeper',
  'Defender',
  'Midfielder',
  'Forward',
];

// Background options
export const SAMPLE_BACKGROUNDS = [
  'Classic Green',
  'Royal Blue',
  'Crimson Red',
  'Golden Yellow',
  'Dark Purple',
];

/**
 * Generate random card data
 */
export function generateRandomCardData(
  overrides: Partial<typeof SAMPLE_PLAYERS.TEST_PLAYER> = {},
) {
  const randomClub =
    SAMPLE_CLUBS[Math.floor(Math.random() * SAMPLE_CLUBS.length)];
  const randomNationality =
    SAMPLE_NATIONALITIES[
      Math.floor(Math.random() * SAMPLE_NATIONALITIES.length)
    ];
  const randomLeague =
    SAMPLE_LEAGUES[Math.floor(Math.random() * SAMPLE_LEAGUES.length)];
  const randomPosition =
    SAMPLE_POSITIONS[Math.floor(Math.random() * SAMPLE_POSITIONS.length)];

  const defence = Math.floor(Math.random() * 100) + 1;
  const control = Math.floor(Math.random() * 100) + 1;
  const attack = Math.floor(Math.random() * 100) + 1;
  return {
    name: `Test Player ${Math.floor(Math.random() * 1000)}`,
    club: randomClub,
    nationality: randomNationality,
    league: randomLeague,
    position: randomPosition,
    stats: { defence, control, attack },
    rating: Math.round((defence + control + attack) / 3),
    ...overrides,
  };
}

/**
 * Generate multiple random cards
 */
export function generateMultipleCards(
  count: number,
): (typeof SAMPLE_PLAYERS.TEST_PLAYER)[] {
  return Array.from({ length: count }, () => generateRandomCardData());
}

/**
 * Card data builder for fluent test data creation
 */
export class CardDataBuilder {
  private data: typeof SAMPLE_PLAYERS.TEST_PLAYER;

  constructor(
    baseData: typeof SAMPLE_PLAYERS.TEST_PLAYER = SAMPLE_PLAYERS.TEST_PLAYER,
  ) {
    this.data = { ...baseData };
  }

  withName(name: string): CardDataBuilder {
    this.data.name = name;
    return this;
  }

  withClub(club: string): CardDataBuilder {
    this.data.club = club;
    return this;
  }

  withNationality(nationality: string): CardDataBuilder {
    this.data.nationality = nationality;
    return this;
  }

  withLeague(league: string): CardDataBuilder {
    this.data.league = league;
    return this;
  }

  withPosition(position: string): CardDataBuilder {
    this.data.position = position;
    return this;
  }

  withStats(stats: {
    defence: number;
    control: number;
    attack: number;
  }): CardDataBuilder {
    this.data.stats = { ...stats };
    return this;
  }

  withRandomStats(): CardDataBuilder {
    this.data.stats = {
      defence: Math.floor(Math.random() * 100) + 1,
      control: Math.floor(Math.random() * 100) + 1,
      attack: Math.floor(Math.random() * 100) + 1,
    };
    return this;
  }

  build(): typeof SAMPLE_PLAYERS.TEST_PLAYER {
    return { ...this.data };
  }
}
